src/pymodaq_plugins_daqmx/hardware/national_instruments/daqmxni.py

```diff
@@ -0,0 +1,729 @@
+        self.counter_type = "Clock Output"
+
+
+class DigitalChannel(Channel):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
+class DOChannel(DigitalChannel):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
+class DIChannel(DigitalChannel):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
+class DAQmx:
+    """Wrapper around the NIDAQmx package giving an easy-to-use object to instantiate channels and tasks"""
+    def __init__(self):
+        self.devices = []
+        self.channels = []
+        self._device = None
+        self._task = None
+        self.update_NIDAQ_devices()
+        self.update_NIDAQ_channels()
+        self.c_callback = None  # A qui servent ces callback ??
+        self.callback_data = None
+        self.is_scalar = True
+        self.write_buffer = np.array([0.])  # ou est utilisé ce buffer??
+
+    @property
+    def task(self):
+        return self._task
+
+    @property
+    def device(self):
+        return self._device
+
+    @device.setter
+    def device(self, device):
+        if device not in self.devices.device_names:
+            raise IOError(f'your device: {device} is not known or connected')
+        self._device = device
+
+    def update_NIDAQ_devices(self):
+        self.devices = self.get_NIDAQ_devices()
+
+    @classmethod
+    def get_NIDAQ_devices(cls):
+        """Get list of NI connected devices
+
+        Returns
+        -------
+        list
+            list of devices as strings to be used in subsequent commands
+        """
+        try:
+            devices = nidaqmx.system.System.local().devices
+            if devices == ['']:
+                devices = []
+            return devices
+        except DaqError as e:
+            return e.error_code
+
+    def update_NIDAQ_channels(self, source_type=None):
+        self.channels = self.get_NIDAQ_channels(self.devices.device_names, source_type=source_type)
+
+    @classmethod
+    def get_NIDAQ_channels(cls, devices=None, source_type=None):
+        """Get the list of available channels for all NiDAq connected devices
+
+        Parameters
+        ----------
+        devices: list
+                 list of strings, each one being a connected device
+        source_type: str
+                     One of the entries of DAQ_NIDAQ_source enum
+
+        Returns
+        -------
+        List of str containing device and channel names
+
+        """
+        if devices is None:
+            devices = cls.get_NIDAQ_devices().device_names
+
+        if source_type is None:
+            source_type = DAQ_NIDAQ_source.names()
+        if not isinstance(source_type, list):
+            source_type = [source_type]
+        channels_tot = []
+        channels = []
+        if not not devices:
+            for device in devices:
+                for source in source_type:
+                    if source == DAQ_NIDAQ_source['Analog_Input'].name:  # analog input
```

Comment:
by the way enums are actually more conveniently used with:
````
DAQ_NIDAQ_source.Analog_Input.name
```
like this you're sure to not call things that are not member of the enum (but again here because the enum already exist in the nidaqmx module, one should directly use it. I did all this to transform at the time the not really pythonic pydaqmx